drivers/mtd_spi_nor/mtd_spi_nor.c

```diff
@@ -303,6 +306,24 @@ static uint32_t mtd_spi_nor_get_size(const mtd_jedec_id_t *id)
         (id->device[1] & ~0x3) == 0) {
         return (0x1F & id->device[0]) * MBIT_AS_BYTES;
     }
+    if (mtd_spi_manuf_match(id, SPI_NOR_JEDEC_MICROCHIP)) {
+        switch (id->device[1]) {
+        case 0x12:  /* SST26VF020A */
+        case 0x8c:  /* SST25VF020B */
+            return  2 * MBIT_AS_BYTES;
+        case 0x54:  /* SST26WF040B */
+        case 0x8d:  /* SST25VF040B */
+            return  4 * MBIT_AS_BYTES;
+        case 0x58:  /* SST26WF080B */
+        case 0x8e:  /* SST25VF080B */
+            return  8 * MBIT_AS_BYTES;
+        case 0x42:  /* SST26VF032B */
+            return 32 * MBIT_AS_BYTES;
+        case 0x43:  /* SST26VF064B */
+        case 0x53:  /* SST26WF064C */
+            return 64 * MBIT_AS_BYTES;
+        }
+    }
 
     /* everyone else seems to use device ID 2 for density */
     return 1 << id->device[1];
```

Comment:
Yeah, the Microchip approach indeed looks so much more elegant! Why should obtaining the flash size should be implemented in an easy, ROM efficient and future proof way when we can have a fun look up table instead?